consensus/polybft/consensus_runtime.go

```diff
@@ -276,6 +276,11 @@ func (c *consensusRuntime) FSM() (*fsm, error) {
 		return nil, err
 	}
 
+	eventRoot, err := c.getExitEventRootHash(epoch.Number)
```

Comment:
Is it before block exsecution or after?

Reply:
Before

Reply:
Could you make event root calculation after block execution?

Reply:
We need to set `eventRoot` before `BuildProposal` is called, because that's where we are creating `Extra` and writing checkpoint data to it.
Why do you see that we need to calculate event root after block execution? I believe it is just a matter of question whether we are including exit events in the block N or block N+1?

Reply:
Fix: Fix: https://github.com/0xPolygon/polygon-edge/pull/838/commits/a6cbda397c6a46283dc10fab61f673a0e1de169c